hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -3,18 +3,1146 @@
+      tests: []
+    }
+    {
+      name: chip_usb_pullup
+      desc: '''Verify the enablement of USB device pullup.
+
+            TBD.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // PINMUX (pre-verified IP) integration tests:
+    {
+      name: chip_pin_mux
+      desc: '''Verify the MIO muxing at input and output sides.
+
+            SW programs MIO INSEL and OUTSEL CSRs to connect and verify each muxed source. At the
+            moment, GPIOs are the only mux inputs.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sleep_pin_mio_dio_val
+      desc: '''Verify the MIO output values in deep sleep state.
+
+            SW programs the MIO OUTSEL CSRs to to ensure that in deep sleep it randomly picks
+            between tie-0, tie-1 or hi-Z for all muxed outputs coming from non-AON IPs. If an AON
+            peripheral output is muxed, then that peripheral's output is selected to ensure in deep
+            sleep the peripheral can continue its signaling even in deep sleep. The testbench
+            verifies the correctness of the reflected values once the chip goes into deep sleep.
+            This is replicated for DIO pins as well.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sleep_pin_wake
+      desc: '''Verify pin wake up from deep sleep state.
+
+            Verify one of the 8 possible MIO or DIO pad inputs (randomly configured) can cause the
+            chip to wake up from sleep state. Verifying wake on posedge is sufficient for the chip
+            level integration testing. Upon wake up, SW reads the wake cause CSR to verify
+            correctness.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // PADCTRL tests:
+    {
+      name: chip_padctrl_attributes
+      desc: '''Verify pad attribute settings for all MIO and DIO pads.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // RV_DM (JTAG) tests:
+    {
+      name: chip_rv_dm_cpu_debug_mem
+      desc: '''Verify access to the debug mem from the CPU.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_rv_dm_jtag_debug_mem
+      desc: '''Verify access to the debug mem from the external JTAG interface.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_rv_dm_cpu_debug_req
+      desc: '''Verify debug request to Ibex while it is actively executing.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_rv_dm_ndm_reset_req
+      desc: '''Verify non-debug reset req initiated from RV_DM when the chip is awake.
+
+            Read CSRs / mem within all IPs in the chip to ensure that they are reset to the original
+            values. Read CSRs / mem in the debug domain to ensure that the values survive the reset.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sleep_rv_dm_ndm_reset_req
```

Comment:
This is a good question... but as currently defined, this reset doesn't work in low power. 
Especially since i think rv_dm isn't even part of the always on power domain.  

We should discuss with the software / nuvoton to make sure they are okay with this. But I tend to think it's okay to keep this off the main power domain, and rely on pinmux to wake the chip up, and then issue a reset if it's needed.